src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/InteropServices/DisableRuntimeMarshallingTests.cs

```diff
@@ -0,0 +1,1072 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Testing;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.InteropServices.DisableRuntimeMarshallingAnalyzer,
+    Microsoft.NetCore.Analyzers.InteropServices.CSharpDisableRuntimeMarshallingFixer>;
+
+namespace Microsoft.CodeAnalysis.NetAnalyzers.UnitTests.Microsoft.NetCore.Analyzers.InteropServices
+{
+    public class DisableRuntimeMarshallingTests
+    {
+        [Fact]
+        public async Task PInvokeWithSetLastError_Emits_Diagnostic()
+        {
+            string source = @"
+using System.Runtime.InteropServices;
+using System.Runtime.CompilerServices;
+
+[assembly:DisableRuntimeMarshalling]
+
+class Foo
+{
+    [DllImport(""Foo"", SetLastError = true)]
+    public static extern void {|CA1420:Bar|}();
```

Comment:
Based on the suggested change in https://github.com/dotnet/roslyn-analyzers/pull/5822/files#r817552866, writing tests for CA1420 would become more tricky when you have the analyzer reporting multiple descriptors with the same diagnostic ID but different messages. @sharwell do you have an example showing the recommended format for writing the interpolation arg for such tests? Or will the correct format continue to work and test library will just pick the first descriptor to verify only the reported diagnostic ID?